src/sir_kibble.c

```diff
@@ -0,0 +1,309 @@
+#include "data.h"
+#include "random.h"
+#include "kirby.h"
+#include "functions.h"
+
+void sub_080B2160(struct Object2 *arg0);
+void sub_080B2210(struct Object2 *arg0);
+void* CreateSirKibble(struct Object* arg0, u8 arg1) {
+    struct Object2 *obj, *obj2;
+    struct Task* task = TaskCreate(ObjectMain, sizeof(struct Object2), 0x1000, TASK_USE_EWRAM, ObjectDestroy);
+    obj2 = TaskGetStructPtr(task);
+    obj = obj2;
+    InitObject(obj, arg0, arg1);
+    if (obj->base.x > obj->kirby3->base.base.base.x) {
+        obj->base.flags |= 1;
+    }
+    sub_0803E2B0(&obj->base, -5, 0, 5, 8);
+    sub_0803E308(&obj->base, -7, -7, 7, 10);
+    ObjectInitSprite(obj);
+    switch (arg0->subtype1) {
+    default:
+    case 0:
+    case 1:
+        sub_080B2160(obj);
+        break;
+    case 2:
+        sub_080B2210(obj);
+        break;
+    }
+    obj->unk9E = 0;
+    obj->unk7C = 0;
+    return obj;
+}
+
+void sub_080B132C(struct ObjectBase *arg0) {
+    if (arg0->xspeed < 0) {
+        arg0->xspeed += 0xe;
+        if (arg0->xspeed >= 1) {
+            arg0->xspeed = 0;
+        }
+    }
+    else {
+        arg0->xspeed -= 0xe;
+        if (arg0->xspeed <= -1) {
+            arg0->xspeed = 0;
+        }
+    }
+    arg0->counter++;
+}
+
+void sub_080B21D8(struct ObjectBase *arg0);
+void sub_080B21A0(struct ObjectBase *arg0);
+void sub_080B1368(struct Object2 *arg0) {
+    arg0->base.flags |= 4;
+    if (arg0->object->subtype1 == 0) {
+        if (arg0->base.xspeed < 0) {
+            arg0->base.xspeed += 0xe;
+            if (arg0->base.xspeed > 0) {
+                arg0->base.xspeed = 0;
+            }
+        }
+        else {
+            arg0->base.xspeed -= 0xe;
+            if (arg0->base.xspeed < 0) {
+                arg0->base.xspeed = 0;
+            }
+        }
+    }
+    else if (arg0->base.unk62 & 3) {
+        arg0->base.xspeed = -arg0->base.xspeed;
+        arg0->kirby3 = sub_0803D368(&arg0->base);
+        if (arg0->base.x > arg0->kirby3->base.base.base.x) {
+            arg0->base.flags  |= 1;
+        }
+        else {
+            arg0->base.flags &= ~1;
+        }
+    }
+
+    if ((arg0->base.unk62 & 4) == 0) {
+        arg0->base.yspeed -= 0x10;
+        if (-0x300 > arg0->base.yspeed) {
+            arg0->base.yspeed = -0x300;
+        }
+
+    }
+    if (arg0->base.counter > 0xb4 - arg0->subtype * 60) {
```

Comment:
```suggestion
    if (arg0->base.counter > 180 - arg0->subtype * 60) {
```
Decimal is more readable here